src/services/index.ts

```diff
@@ -188,37 +206,94 @@ export function deployAMBModule(
   ];
 }
 
+export function deployCirculatingSupplyContract(
+  chainId: number,
+  circulatingSupply: BigNumberish,
+  saltNonce: string
+) {
+  const provider = getProvider(chainId);
+  const circulatingSupplyContract = new ethers.Contract(
+    CIRCULATING_SUPPLY_MASTER_COPY_ADDRESS,
+    CIRCULATING_SUPPLY_CONTRACT_ABI
+  );
+  const { factory } = getFactoryAndMasterCopy("exit", provider, chainId);
+
+  const moduleSetupData =
+    circulatingSupplyContract.interface.encodeFunctionData("setUp", [
+      circulatingSupply,
+    ]);
+
+  const expectedAddress = calculateProxyAddress(
+    factory,
+    circulatingSupplyContract.address,
+    moduleSetupData,
+    saltNonce
+  );
+
+  const deployData = factory.interface.encodeFunctionData("deployModule", [
+    circulatingSupplyContract.address,
+    moduleSetupData,
+    saltNonce,
+  ]);
```

Comment:
The CirculatingSupply contract is deployed from the ModuleFactory.
after the deployment, the `ModuleProxyCreation` event is emited, I'm not sure if having non-module deployments could affect us somehow. could this cause issues/concerns in the future?

Reply:
The safe can't deploy contracts, so we have to deploy contracts from other contracts